media/windows/ECC-Ecobee/ECC Pycobee Data.py

```diff
@@ -2326,50 +2410,95 @@ def get_snapshot(conn,
     return lists_dict  # Dictionary with more data to process in lists
 
 
-def check_internet_connect(site_url):
+def check_internet_connect(url_root):
```

Comment:
Minor nitpick: this isn't actually a URL: it's a hostname.

FWIW, you might want to make the hosts that are checked transparent to the caller -- just have them call `check_internet_connect()`, and it returns `True` or `False`.  Inside `check_internet_connect()`, it does some magic stuff to check internet connectivity (the caller doesn't care what it does, just that it checks if internet connectivity is up or down).

Also, you might want to do a "short" internet check.  https://google.com/ is pretty famous for being a very short `index.html` (i.e., download).  It's currently clocking in at about 11K.   Just for comparison, https://ecobee.com/ is clocking in at around 390K.

Honestly, 390K probably doesn't matter to us with our internet connectivity.  But it still is kind of a heavyweight download.

Reply:
Yah...
I know it is a hostname, not a URL.  This is an artifact from a prior iteration of the code where I WAS passing a URL.  In later versions I changed this to a hostname as I added other functionality.  Just didn't go back and change the variable names.

Making the hosts checked transparent to the caller is certainly another approach.  I'd written this routine to be a little more generic, thinking I might use it elsewhere in the future and didn't want to always check the same site(s), so made it part of the argument list.

Regarding the size of the check (ecobee.com), nice to know it's large, but I agree it probably doesn't matter here.  I wouldn't want to drop it for ecobee in this app, as it is the main site that is hit to poll data.  Again, this function is only invoked in rare instances where repeated time-out errors have occurred...so the impact should be trivial.

Reply:
Roger.  Don't get me wrong -- I don't have strong feelings about this 😄, and our internet connectivity can certainly handle these downloads.

FWIW: it doesn't sound like it's an issue in this case, but be aware that web sites with "heavy" downloads sometimes have DDOS protections in place.  I.e., if they see frequent downloads from a single IP, they may temporarily block traffic from that IP.  Doesn't sound like this is going to be an issue here; just something to keep in mind.

Reply:
Got it...makes sense, thanks!